sycl/tools/sycl-xocc/bin/sycl-xocc

```diff
@@ -60,58 +61,86 @@ if [[ -z "$6" ]]; then
   usage 6 "no final output file specified"
 fi
 
-# Hardcoded path from the bin directory that xocc is stored in, works for now 
-# but a future TODO might be to allow this to be optionally specified as an 
-# ENV variable 
-XOCC_LIB_SPIR="$1/../lnx64/lib/libspir64-39-hls.bc"
+# Setup SDx related env variables
+SDX_BIN_PATH_DIR="$1"
+if [[ ! -d "$SDX_BIN_PATH_DIR" ]]; then
+  usage 7 "path to SDx bin directory is not a valid directory"
+else
+  # Hardcoded path from the bin directory that xocc is stored in, works for now
+  # but a future TODO might be to allow this to be optionally specified as an
+  # ENV variable
+  XOCC_LIB_SPIR="$SDX_BIN_PATH_DIR/../lnx64/lib/libspir64-39-hls.bc"
+  if [[ ! -f "$XOCC_LIB_SPIR" ]]; then
+    usage 8 "could not find SDx's libspir64-39-hls.bc"
+  fi
+
+  XOCC="$SDX_BIN_PATH_DIR/xocc"
+  if [[ ! -f "$XOCC" ]]; then
+    usage 8 "could not find xocc executable in $SDX_BIN_PATH_DIR"
+  fi
+fi
 
-# Perhaps not so important that the linker is the one packaged with the Clang
-# build, but it's important that Opt is so that it's packaged with the correct 
-# LLVM passes. 
-OPT="$2/opt"
-LLVM_LINK="$2/llvm-link"
-XOCC="$1/xocc"
+# Setup Clang/LLVM related env variables
+DRIVER_PATH_DIR="$2"
+if [[ ! -d "$DRIVER_PATH_DIR" ]]; then
+  usage 8 "path to driver directory is not a valid directory"
+else
+  # Perhaps not as important that the linker is the one packaged with the Clang
+  # build (although if the linker version is too different it can inject incorrect
+  # IR during the link process), but it's very important that Opt is so that it's
+  # packaged with the correct LLVM passes.
+  OPT="$DRIVER_PATH_DIR/opt"
+  if [[ ! -f "$OPT" ]]; then
+    usage 8 "no llvm opt executable found in $DRIVER_PATH_DIR"
+  fi
+
+  LLVM_LINK="$DRIVER_PATH_DIR/llvm-link"
+  if [[ ! -f "$LLVM_LINK" ]]; then
+    usage 8 "no llvm-link executable found in $LLVM_LINK"
+  fi
+fi
 
 $OPT -O3 -asfix -globaldce -inSPIRation -globaldce -kernelPropGen "$4" -o \
   "$5/$3_kernels-optimized.bc"
```

Comment:
I have the feeling that here and in the following we should have some symbolic names instead of `$3`, `$4` and `$5`